src/FrankaLightWeightInterface.cpp

```diff
@@ -64,11 +85,25 @@ void FrankaLightWeightInterface::run_controller() {
 
 void FrankaLightWeightInterface::poll_external_command() {
   if (poll(this->zmq_command_msg_)) {
-    this->command_joint_torques_ = Eigen::Map<Eigen::MatrixXd>(this->zmq_command_msg_.jointTorque.data.data(), 7, 1);
     this->last_command_ = std::chrono::steady_clock::now();
+    this->control_type_ = this->zmq_command_msg_.controlType;
+
+    this->command_joint_positions_ = Eigen::Map<Eigen::MatrixXd>(this->zmq_command_msg_.jointPosition.data.data(), 7, 1);
+    this->command_joint_velocities_ = Eigen::Map<Eigen::MatrixXd>(this->zmq_command_msg_.jointVelocity.data.data(), 7, 1);
+    this->command_joint_torques_ = Eigen::Map<Eigen::MatrixXd>(this->zmq_command_msg_.jointTorque.data.data(), 7, 1);
+
+    this->command_cartesian_position_ = Eigen::Vector3d(this->zmq_command_msg_.eePose.position.data().data());
+    this->command_cartesian_orientation_ = Eigen::Quaterniond(this->zmq_command_msg_.eePose.orientation.data().data());
+    this->command_cartesian_twist_.block<3, 1>(0, 0) = Eigen::Vector3d(this->zmq_command_msg_.eeTwist.linear.data().data());
+    this->command_cartesian_twist_.block<3, 1>(3, 0) = Eigen::Vector3d(this->zmq_command_msg_.eeTwist.angular.data().data());
+    this->command_cartesian_wrench_.block<3, 1>(0, 0) = Eigen::Vector3d(this->zmq_command_msg_.eeWrench.linear.data().data());
+    this->command_cartesian_wrench_.block<3, 1>(3, 0) = Eigen::Vector3d(this->zmq_command_msg_.eeWrench.angular.data().data());
   } else if (std::chrono::duration_cast<std::chrono::milliseconds>(
       std::chrono::steady_clock::now() - this->last_command_).count() > this->command_timeout_.count()) {
+    this->command_joint_velocities_.setZero();
```

Comment:
Why don't you set the command_joint_positions, command_cartesian_position and -quaternion to zero here?

Reply:
Because you want the robot to stay where it is. Not go to the zero joint or Cartesian position